src/main/java/org/spongepowered/common/world/SimulatedPlayer.java

```diff
@@ -0,0 +1,127 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.world;
+
+import com.mojang.authlib.GameProfile;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.network.EnumPacketDirection;
+import net.minecraft.network.NetHandlerPlayServer;
+import net.minecraft.network.NetworkManager;
+import net.minecraft.network.Packet;
+import net.minecraft.network.play.client.CPacketClientSettings;
+import net.minecraft.server.management.PlayerInteractionManager;
+import net.minecraft.stats.StatBase;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.world.WorldServer;
+import org.spongepowered.common.SpongeImpl;
+
+import java.util.Set;
+
+// Based on Forge's FakePlayer
+public class SimulatedPlayer extends EntityPlayerMP implements FakePlayer {
+
+    public static class FakeNetHandler extends NetHandlerPlayServer {
+
+        private static final NetworkManager fakeNetworkManager = new NetworkManager(EnumPacketDirection.CLIENTBOUND);
+
+        public FakeNetHandler(EntityPlayerMP player) {
+            super(player.mcServer, fakeNetworkManager, player);
+        }
+
+        @Override
+        public void update() {
+        }
+
+        @Override
+        public void disconnect(String reason) {
+        }
+
+        @Override
+        public void setPlayerLocation(double x, double y, double z, float yaw, float pitch, @SuppressWarnings("rawtypes") Set relativeSet) {
+        }
+
+        @Override
+        public void onDisconnect(ITextComponent reason) {
+        }
+
+        @Override
+        public void sendPacket(Packet<?> packetIn) {
+        }
+    }
+
+    public SimulatedPlayer(WorldServer world, GameProfile profile) {
+        super(SpongeImpl.getServer(), world, profile, new PlayerInteractionManager(world));
+        this.connection = new FakeNetHandler(this);
+    }
+
+    @Override
+    public boolean canUseCommand(int i, String s) {
+        return false;
+    }
+
+    @Override
+    public void sendMessage(ITextComponent component) {
+    }
+
+    @Override
+    public void sendStatusMessage(ITextComponent chatComponent, boolean actionBar) {
+    }
+
+    @Override
+    public void addStat(StatBase par1StatBase, int par2) {
+    }
+
+    @Override
+    public boolean isEntityInvulnerable(DamageSource source) {
+        return true;
+    }
+
+    @Override
+    public boolean canAttackPlayer(EntityPlayer player) {
+        return false;
+    }
+
+    @Override
+    public void onDeath(DamageSource source) {
+        return;
+    }
+
+    @Override
+    public void onUpdate() {
+        return;
+    }
+
+    @Override
+    public Entity changeDimension(int dim) {
+        return this;
+    }
+
+    @Override
+    public void handleClientSettings(CPacketClientSettings packetIn) {
+    }
+}
```

Comment:
I took it from forge's [FakePlayer](https://github.com/MinecraftForge/MinecraftForge/blob/1.11.x/src/main/java/net/minecraftforge/common/util/FakePlayer.java#L45-L55), I think it covers the main cases. For the restricted amount of actions available, and no external access to the `SimulatedPlayer` there shouldn't be any unexpected method calls. (Note: in spongeforge I use forge's `FakePlayer` so mods can handle fake players properly)